kymatio/scattering2d/frontend/torch_frontend.py

```diff
@@ -0,0 +1,108 @@
+__all__ = ['Scattering2DTorch']
+
+import torch
+
+from .base_frontend import Scattering2DBase
+from kymatio.scattering2d.core.scattering2d import scattering2d
+from ...frontend.torch_frontend import ScatteringTorch
+
+
+class Scattering2DTorch(ScatteringTorch, Scattering2DBase):
+    def __init__(self, J, shape, L=8, max_order=2, pre_pad=False, backend='torch'):
+        ScatteringTorch.__init__(self)
+        Scattering2DBase.__init__(**locals())
```

Comment:
This passes in all the __init__ params. 

[locals()](https://www.programiz.com/python-programming/methods/built-in/locals) essentially gives you a dictionary of all the local varaibles in the current scope of the function (here the dictionary would include the parameters passed into __init__). 

The [**](https://stackoverflow.com/a/36926) allows us to pass multiple arguments from a dictionary into a function. 

Do you think this is too pythonic/complicated? I was passing in the parameters manually before hand, but that looked a bit messy.

Reply:
While this sounds like a good idea to factorize the code.. I think this is a bit dangerous because it passes some additional variables: you can imagine unfortunate conflict..

Reply:
Yeah this is dangerous (and, I would argue *un*pythonic). Better to spell things out here.